interpreter/types/adapters_test.go

```diff
@@ -0,0 +1,137 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"bytes"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/struct"
+	"github.com/google/cel-go/interpreter/types/aspects"
+	expr "github.com/google/cel-spec/proto/v1/syntax"
+	"reflect"
+	"testing"
+)
+
+func TestExprToProto(t *testing.T) {
+	// Core type conversion tests.
+	expectExprToProto(t, true, true)
+	expectExprToProto(t, int64(-1), int32(-1))
+	expectExprToProto(t, int64(2), int64(2))
+	expectExprToProto(t, uint64(3), uint32(3))
+	expectExprToProto(t, uint64(4), uint64(4))
+	expectExprToProto(t, float64(5.5), float32(5.5))
+	expectExprToProto(t, float64(-5.5), float64(-5.5))
+	expectExprToProto(t, "hello", "hello")
+	expectExprToProto(t, []byte("world"), []byte("world"))
+	expectExprToProto(t, []int64{1, 2, 3}, []int32{1, 2, 3})
+	expectExprToProto(t, NewListValue([]int64{1, 2, 3}), []int32{1, 2, 3})
+	expectExprToProto(t, map[int64]int64{1: 1, 2: 1, 3: 1},
+		map[int32]int32{1: 1, 2: 1, 3: 1})
+	expectExprToProto(t, NewMapValue(map[int64]int64{1: 1, 2: 1, 3: 1}),
+		map[int32]int32{1: 1, 2: 1, 3: 1})
+
+	// Null conversion tests.
+	expectExprToProto(t, structpb.NullValue_NULL_VALUE, structpb.NullValue_NULL_VALUE)
+
+	// Proto conversion tests.
+	parsedExpr := &expr.ParsedExpr{}
+	expectExprToProto(t, NewProtoValue(parsedExpr), parsedExpr)
+	expectExprToProto(t, NewProtoValue(*parsedExpr), *parsedExpr)
+}
+
+func TestProtoToExpr(t *testing.T) {
+	// Core type conversions.
+	expectProtoToExpr(t, true, true)
+	expectProtoToExpr(t, int32(-1), int64(-1))
+	expectProtoToExpr(t, int64(2), int64(2))
+	expectProtoToExpr(t, uint32(3), uint64(3))
+	expectProtoToExpr(t, uint64(4), uint64(4))
+	expectProtoToExpr(t, float32(5.5), float64(5.5))
+	expectProtoToExpr(t, float64(-5.5), float64(-5.5))
+	expectProtoToExpr(t, "hello", "hello")
+	expectProtoToExpr(t, []byte("world"), []byte("world"))
+	expectProtoToExpr(t, []int32{1, 2, 3},
+		NewListValue([]int32{1, 2, 3}))
+	expectProtoToExpr(t, map[int32]int32{1: 1, 2: 1, 3: 1},
+		NewMapValue(map[int32]int32{1: 1, 2: 1, 3: 1}))
+
+	// Null conversion test.
+	expectProtoToExpr(t, structpb.NullValue_NULL_VALUE, structpb.NullValue_NULL_VALUE)
+
+	// Proto conversion test.
+	parsedExpr := &expr.ParsedExpr{}
+	expectProtoToExpr(t, parsedExpr, NewProtoValue(parsedExpr))
+}
+
+func TestUnsupportedConversion(t *testing.T) {
+	if val, err := ProtoToExpr(NonConvertiable{}); err == nil {
+		t.Error("Expected error when converting non-proto struct to proto", val)
+	}
+	if val, err := ExprToProto(reflect.TypeOf(NonConvertiable{}), NonConvertiable{}); err == nil {
+		t.Error("Expected error when converting non-proto expr to proto", val)
+	}
+}
+
+func expectExprToProto(t *testing.T, in interface{}, out interface{}) {
```

Comment:
The Google Go style is to avoid assert libraries like these two expect*() functions. (Internally see http://go/gocomments#TOC-Assert)

Reply:
The helper (marked as such with t.Helper()) seems to clarify the nature of the test. It primarily exists as a comparison helper which is one area where the assert discussion seems to talk about needing to handle comparison of got versus wanted via libraries.